[ADD] estate: created new models following Odoo Guide t#70762

Cindypsd opened the pull request:
**Explanation**

I created a new Odoo module called Real Estate by following the guidelines laid out in the Odoo Guide chapter 1 to 8. The module includes 4 models that are designed to summarize important information for properties.

![Captura de Pantalla 2023-04-24 a la(s) 18 05 44](https://user-images.githubusercontent.com/108036643/234140861-188dba58-f7fc-4849-8bdf-1558050b7526.png)

This module allows users to access a comprehensive description of each property, and it also enables them to add tags and categorize properties by type.

![Captura de Pantalla 2023-04-24 a la(s) 17 55 14](https://user-images.githubusercontent.com/108036643/234139990-b22f4ce9-2186-4ee5-be8f-69698ddf0fce.png)

Users can view offers made for each property along with the salesperson's details responsible for selling it.

![Captura de Pantalla 2023-04-24 a la(s) 17 55 23](https://user-images.githubusercontent.com/108036643/234140135-0fb5e2e0-068a-4f4e-a233-d5e700a381fc.png)
![Captura de Pantalla 2023-04-24 a la(s) 17 55 31](https://user-images.githubusercontent.com/108036643/234140142-0bcae89f-6681-4ec9-85e9-2012bf371348.png)

The module also enables users to search for properties, apply filters and group the results as per their preferences.

![Captura de Pantalla 2023-04-24 a la(s) 17 59 12](https://user-images.githubusercontent.com/108036643/234140238-87aea367-05ea-42c3-926a-8c91f4749162.png)

sebasdrk17 commented on estate/models/__pycache__/__init__.cpython-311.pyc:
pyc files should not be loaded, add them to .git ignore please

sebasdrk17 commented on estate/models/estate_property.py:
```diff
@@ -5,7 +5,7 @@
 class EstateProperty(models.Model):
     _name = "property"
     _description = "Test Model for Training"
```
Use a propper description for the model please

what does it do?

sebasdrk17 commented on estate/models/estate_property.py:
```diff
@@ -5,7 +5,7 @@
-   
+
     name = fields.Char(string = 'Name', required=True)
     description = fields.Text(string = 'Description')
```
No need for "string" here, since is the same as the field name the result will be the same

For example:

"description_for_property" the resulting string will be "Description For Property"

remove the string please

```suggestion
    description = fields.Text()
```

sebasdrk17 commented on estate/models/estate_property.py:
```diff
@@ -5,7 +5,7 @@
+
     name = fields.Char(string = 'Name', required=True)
     description = fields.Text(string = 'Description')
     postcode = fields.Char(string = 'Postcode')
```
Remove the string, it is not needed, do the same for all the fields that have a string and is not needed

for example:

"postcode" will be parsed to "Postcode" so the string is not needed at all

let's say we have this field:

"property_state_code_ids" it is a technical name, in that case we want to simplify what it means so:

property_state_code_ids = fields....(string = 'Property region code')

In that case is valid because we want to simplify the string of the field

sebasdrk17 replied:
I cannot review on  lines 12 to 43,

so ill leave the comments here:

https://github.com/Cindypsd/Odoo_Training/pull/1/files#diff-26167efdfcbe961332e29e17ced980966bad910db825ec57c717a38b0b591d14R16

this is a good example for when string is needed

comments:

1. in this [line](https://github.com/Cindypsd/Odoo_Training/pull/1/files#diff-26167efdfcbe961332e29e17ced980966bad910db825ec57c717a38b0b591d14R21)

the first parameter of the selection is the technical name the selection will have, change all the first parameter to be more technical (lowercase) e.g. north,south,....

sebasdrk17 commented on estate/models/estate_property.py:
```diff
@@ -44,8 +44,12 @@ class EstateProperty(models.Model):
     property_type_id = fields.Many2one("property.type", string="Property Type")
```
string not needed

```suggestion
    property_type_id = fields.Many2one("property.type")
```

sebasdrk17 replied:
missing point here

sebasdrk17 commented on estate/models/estate_property.py:
```diff
@@ -44,8 +44,12 @@ class EstateProperty(models.Model):
+        string="Salesman",
         default=lambda self: self.env.user.id,
     )
+    buyer_id = fields.Many2one("res.partner", string="Buyer", copy=False)
+    tag_ids = fields.Many2many("property.tag", string="Tags")
```
strings not needed, remove them please

```suggestion
    buyer_id = fields.Many2one("res.partner", copy=False)
    tag_ids = fields.Many2many("property.tag")
```

sebasdrk17 replied:
missing point here

sebasdrk17 commented on estate/models/estate_property_offer.py:
```diff
@@ -0,0 +1,18 @@
+
+class PropertyOffer(models.Model):
+    _name = "property.offer"
+    _description = "Property Offer"
```
Add a better description please

sebasdrk17 commented on estate/models/estate_property_offer.py:
```diff
@@ -0,0 +1,18 @@
+    _description = "Property Offer"
+
+    price = fields.Float()
+    status = fields.Selection(
```
string not needed

sebasdrk17 commented on estate/models/estate_property_offer.py:
```diff
@@ -0,0 +1,18 @@
+        copy=False,
+    )
+    partner_id = fields.Many2one("res.partner", required=True)
+    property_id = fields.Many2one("property", string="Property", required=True)
```
string not needed

sebasdrk17 replied:
missing comment here

sebasdrk17 commented on estate/models/estate_property_tag.py:
```diff
@@ -0,0 +1,8 @@
+
+class PropertyTag(models.Model):
+    _name = "property.tag"
+    _description = "Property Tags"
```
Use better description please

sebasdrk17 commented on estate/models/estate_property_tag.py:
```diff
@@ -0,0 +1,8 @@
+    _name = "property.tag"
+    _description = "Property Tags"
+
+    name = fields.Char(string= 'Name', required=True)
```
string not needed

sebasdrk17 commented on estate/views/estate_property_tag_views.xml:
```diff
@@ -0,0 +1,24 @@
+        <form string="Property Tag">
+            <sheet>
+                <group>
+                    <field name="name" string="Name" />
```
String not needed

sebasdrk17 reviewed (commented):
General points:

1. If the string is exactly the same as the name of the field, then string is not needed, therefore string needs to be removed
2. MR is lacking information, what does this PR addresses? why is it needed? what changed?
3. Related to point 2, what do I need to test? what should I be seeing in the module?
4. Everything needs to be in English (code, MR...), only internal communication can be in Spanish

So far is looking great, needs some details but good first PR

Good job!

Cindypsd commented:
@sebasdrk17 

I updated the pull request to address the following issues raised in the code review:

Removed unnecessary strings that were identical to the field names.
Resolved conflicts in the files to be merged.
Added .pyc files to .gitignore to ensure they are not loaded.
Improved the description for the models to be more informative.

Overall, the Estate Property model now accurately tracks the property's status and provides fields to identify the salesperson responsible for selling the property and track potential buyers. It also includes features for categorization and searching, such as assigning a property type and multiple tags. Additionally, the model allows for multiple offers to be made on a single property, each tracked separately in a one-to-many relationship.

I look forward to any comments or feedback you may have. Thank you for taking the time to review the code.

sebasdrk17 commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,59 @@
+class EstateProperty(models.Model):
+    _name = "property"
+    _description = """
+    This model provides a representation of a real estate property, including important fields such as the property name, description, postcode, date availability, expected price, selling price, number of bedrooms, living area, number of facades, and other features such as garage and garden.
```
This description is Okay but its just too long, we can shorten it like this:

```suggestion
Representation of a real estate property
```

The fields itsel are self explanatory so, there is no need to actually telling here all the fields

sebasdrk17 commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,59 @@
+    garden = fields.Boolean()
+    garden_area = fields.Integer()
+    garden_orientation = fields.Selection(
+        selection=[
+            ('North', 'North'),
+            ('South', 'South'),
+            ('East', 'East'),
+            ('West', 'West')
+        ],
```
use lowercase here please for the code representation (left string)

sebasdrk17 commented on estate/models/estate_property_offer.py:
```diff
@@ -0,0 +1,19 @@
+class PropertyOffer(models.Model):
+    _name = "property.offer"
+    _description = """
+    This model represents an offer made by a buyer for a property. It has fields to track the offer price, its status (accepted or refused), and the partner (buyer) who made the offer.
```
description too long, check the comment I've made above

sebasdrk17 commented on estate/models/estate_property_tag.py:
```diff
@@ -0,0 +1,10 @@
+class PropertyTag(models.Model):
+    _name = "property.tag"
+    _description = """
+    This is a model that represents tags that can be associated with different properties.This model can be used in conjunction with the "property" model, as it allows users to add descriptive tags to each property record.
```
description too long

sebasdrk17 commented on estate/models/estate_property_type.py:
```diff
@@ -0,0 +1,10 @@
+
+class PropertyType(models.Model):
+    _name = "property.type"
+    _description = """
```
description too long

sebasdrk17 commented on estate/models/estate_property_type.py:
```diff
@@ -0,0 +1,10 @@
+    """
+
+    name = fields.Char(required=True)
+    property_ids = fields.One2many("property", "property_type_id", "Properties")
```
why do we need "properties"?

when using the postfix"_ids" it will make the label plural

sebasdrk17 reviewed (commented):
1. PR descriptions is still missing, we need to edit the description and update it

Check the image I've sent you in private so that you have an example, also there is a guide on how to document a PR in the trainning document

2. Descriptions are way too long
3. There are still some "strings" in models that needs to be removed

Cindypsd commented:
@sebasdrk17 

I made the requested changes to the models by reducing the length of descriptions and removing any unnecessary code.I am looking forward to your review and feedback on these changes.

sebasdrk17 reviewed (commented):
Code looks good

There is still missing a proper description to your MR, the one that you have right not it's not bad

But a description should state what was the problem, what did you make to solve the problem and video/images of the changes

I've sent you an image and an example of some MR's 

Here are some missing points:

1. Your pull request should also say the number of your task, in this case the task where you are filling your ours
2. There is no visual feedback of what you did, how do I know as a client what changes were made?
3. This line is not needed, please remove it "As of now, the module is ready for testing and I am looking forward to receiving your feedback on it."

sebasdrk17 commented:
LGTM
